nemo-physical/src/table_reader.rs

```diff
@@ -0,0 +1,20 @@
+//! This modules defines the [`TableReader`] trait, which is part of the interface to nmo_logical.
+//!
+
+use crate::{builder_proxy::PhysicalBuilderProxyEnum, error::Error};
+
+/// A general interface for reading tables from files.
+///
+/// This is called from the physical layer to ask a reader to fill the
+/// Vector of [builder proxies][PhysicalBuilderProxyEnum].
+///
+/// # Note
+/// This is the physical interface to access all readers, instantiated on the logical layer.
+/// Therefore every reader needs to implement this trait.
+pub trait TableReader: std::fmt::Debug {
+    /// Read the table into multiple [`ColumnBuilderProxy`][crate::builder_proxy::PhysicalColumnBuilderProxy]
+    fn read_into_builder_proxies<'a: 'b, 'b>(
+        &self,
+        builder_proxies: &'b mut Vec<PhysicalBuilderProxyEnum<'a>>,
+    ) -> Result<(), Error>;
+}
```

Comment:
We might want to have a separate error type for things that could go wrong during reading (since those are shared between the logical and physical layers) – possibly also later on for writing. Even if we don't do this, I guess it might make sense to have the concrete Error type here given as an associated type, but we do require this to be object safe, don't we?

Reply:
I like the new `ReadingError` (we might want to change the name later on, but for now this is good).